src/wallet/wallet.cpp

```diff
@@ -2701,7 +2702,7 @@ OutputType CWallet::TransactionChangeType(const Optional<OutputType>& change_typ
     return m_default_address_type;
 }
 
-bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransactionRef& tx, CAmount& nFeeRet, int& nChangePosInOut, bilingual_str& error, const CCoinControl& coin_control, bool sign)
+bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransactionRef& tx, CAmount& nFeeRet, int& nChangePosInOut, bilingual_str& error, const CCoinControl& coin_control, std::string& fee_reason, bool sign)
```

Comment:
Seems like it would work better to use `FeeReason*` here (or perhaps `FeeCalculation*`)

Reply:
Are you recommending changing the pointer name from fee_reason to FeeReason?

Reply:
He is referring to the enum `FeeReason` and the struct `FeeCalculation` to be used here instead of a string.

Reply:
How would i go about that because i'd have to change everything that calls `createTransaction` and also the UniValue entry string type? I'm just a bit confused because when I change its type to `FeeReason*', I get bombarded with errors

Reply:
There is a bit more to it than just changing the type here in this line. The other places where you pass in a string now you will need to use the `FeeReason*` type as well. You also won't convert the reason to a string in this function but where you actually need it as a string. But most importantly: take your time to and try to understand these errors. Once you understand them they can guide you to the right solution.

Reply:
I changed the `str fee reason` to struct `FeeCalculation`